### PR TITLE
Closes #1619: write up to explain mocks organization and generation

### LINKED ISSUE #1619: include a short write up about how mocks are organized for unit testing
This issue is to track and resolve the comment raised by @astoycos for the PR https://github.com/ovn-org/ovn-kubernetes/pull/1602

### PR DESCRIPTION
**Pull request by vjayaramrh**
Signed-off-by: Vishwanath Jayaraman <vjayaram@redhat.com>

**- What this PR does and why is it needed**

This PR adds documentation that explains how mocks for unit tests are organized and generated.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by vjayaramrh** [comment]
> Would it be possible to either link to or describe how to actually use the mock functions in a testing env?

@JacobTanenbaum great suggestion

File: docs/mocks-ut-faq.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	`go-controller/pkg/util/mocks/ExecRunner.go` file.
+	
+	e.g; b) The `SriovNetLibOps` interface defined in `go-controller/pkg/cni/helper_linux.go` file has its mock 
+	generated in `go-controller/pkg/cni/mocks/SriovNetLibOps.go` file.
+	
+### How are the mocks for interfaces to be consumed by unit tests currently generated?
+
+- The vektra/mockery package from https://github.com/vektra/mockery is leveraged to auto generate mocks for defined interfaces.
+
+- Mocks for interfaces can be generated using vektra/mockery in one of two ways:
```
Comment (trozet) [inline docs/mocks-ut-faq.md:27]: should you now add the option for running the script I just merged :) ?

**Review by trozet — CHANGES_REQUESTED** [review]

File: docs/mocks-ut-faq.md (hunk 1 above)
Comment (vjayaramrh) [reply docs/mocks-ut-faq.md:27]: > should you now add the option for running the script I just merged :) ?

good point, let me add a section on how to use the scripts to regenerate mocks

**Review by trozet — APPROVED** [review]

**Pull request merged**